completion.go

```diff
@@ -10,15 +10,23 @@ import (
 
 const completionFlag = "--mcli-generate-completion"
 
-func hasCompletionFlag(args []string) (bool, []string) {
-	var lastArg string
-	if len(args) > 0 {
-		lastArg = args[len(args)-1]
-	}
-	if lastArg == completionFlag {
-		return true, args[:len(args)-1]
+func getAllowedShells() []string {
+	return []string{"bash", "zsh", "fish", "powershell"}
+}
+
+func hasCompletionFlag(args []string) (bool, []string, string) {
+	shell := "unsupported"
+	foundIndex := find(args, completionFlag)
+	if len(args) >= 2 && foundIndex >= 0 {
+		proposedShell := args[foundIndex+1]
+		if contains(getAllowedShells(), proposedShell) {
+			shell = proposedShell
+			args = remove(args, completionFlag)
+			args = remove(args, shell)
```

Comment:
This may remove user flag value unexpectedly.
Since we have verified the args length and completion flags appearence, simply use "args = args[:len(args)-2]" is guaranteed to be correct.

Reply:
From a design perspective, we could make this be more robust, we can index the `completionFlag`, then check the shell type after if if exsits, if there are any more args, just ignore them. By doing this, it is more friendly for adding more args in future (most probably never~).

Reply:
I am already searching for index of completionFlag, so I will make use of it.